File: linked-list/insertarAlPrincipio/insertAtTheBeginning.ts
// Importamos la definición de la lista enlazada y el nodo desde '../type'
import { LinkedList, Node } from '../type';

// Clase que extiende la lista enlazada con un método para insertar al principio
export class LinkedListWithInsertB extends LinkedList {
  insertAtTheBeginning(data: number): void {
    const newNode = new Node(data); // Creamos un nuevo nodo con el valor 'data'
    newNode.next = this.head; // Enlazamos el nuevo nodo con la cabeza actual
    this.head = newNode; // El nuevo nodo se convierte en la nueva cabeza de la lista
  }
}

File: linked-list/insertarAlPrincipio/insertAtTheBeginning.pas
// Procedimiento para insertar un elemento al principio de una lista enlazada
procedure insertAtTheBeginning(var actual: list; data: integer);
var
    aux: list; // Variable para el nuevo nodo
begin
    new(aux); // Creamos un nuevo nodo
    aux^.data := data; // Asignamos el valor 'data' al campo 'data' del nodo
    aux^.next := actual; // Enlazamos el nuevo nodo con el nodo actual (anterior cabeza)
    actual := aux; // El nuevo nodo se convierte en la nueva cabeza de la lista
end;
